tests/test_metrics.py

```diff
@@ -518,6 +518,79 @@ def test_calc_change_score(self):
         expected_change_score = -2
         self.assertEqual(expected_change_score, calculated_change_score)
 
+    def test_calculate_mple_regressors(self):
+        # directed
+        node_attr = np.array([3, 5, 7])
+        W = np.array([[0, 1, 0],
+                      [1, 0, 1],
+                      [1, 0, 0]])
+        attr_sum_mat = np.array([[0, 8, 10],
+                                 [8, 0, 12],
+                                 [10, 12, 0]])
+
+        metric_both = NodeAttrSum(node_attr, is_directed=True)
+        indices_lims = (0, 6)
+        expected_regressors_1 = np.array([[8.],
+                                          [10.],
+                                          [8.],
+                                          [12.],
+                                          [10.],
+                                          [12.]])
+        self.assertTrue(metric_both.calculate_mple_regressors(W, indices_lims) == expected_regressors_1)
+
+        indices_lims = (1, 4)
+        expected_regressors_2 = np.array([[10.],
+                                          [8.],
+                                          [12.]])
+        self.assertTrue(metric_both.calculate_mple_regressors(W, indices_lims) == expected_regressors_2)
+
+        metric_out = NodeAttrSumOut(node_attr, is_directed=True)
+        metric_in = NodeAttrSumIn(node_attr, is_directed=True)
+        indices_lims = (0, 6)
+
+        self.assertTrue(metric_out.calculate_mple_regressors(W, indices_lims) == expected_regressors_1)
```

Comment:
I don't get why the expected result of `calculate_mple_regressors` is the same for `NodeAttrSum` and for `NodeAttrSumIn`/`NodeAttrSumOut`. Not both of the attributes are summed for each edge...

Reply:
no worries it is not, again was not tested properly, it's ok now